src/app/search/page.tsx

```diff
@@ -0,0 +1,53 @@
+import { redirect } from 'next/navigation';
+import { dehydrate, HydrationBoundary, QueryClient } from '@tanstack/react-query';
+import classNames from 'classnames/bind';
+
+import Pagination from '@/components/Pagination/Pagination';
+import { getSearchResult } from '@/api/searchAPI';
+import { SearchResultType } from '@/types/SearchType';
+import { ROUTER } from '@/constants/route';
+import SearchBox from '@/components/SearchBox/SearchBox';
+import { Title, NoResult, CardList } from './_components';
+
+import styles from './page.module.scss';
+
+const cn = classNames.bind(styles);
+
+interface SearchPageProps {
+  searchParams: Record<string, string>;
+}
+
+export default async function SearchPage({ searchParams }: SearchPageProps) {
+  const keyword = searchParams.keyword ?? '';
+  const page = Number(searchParams.page ?? 0);
```

Comment:
```suggestion
  const { keyword = '', page = '0' } = searchParams;
  const pageNumber = Number(page);
```
사소하지만 여기도 구조분해 할당이 더 나을거같아요